src/server.rs

```diff
@@ -544,6 +544,12 @@ where
         self.connection.go_away_gracefully();
     }
 
+    #[doc(hidden)]
+    #[cfg(feature = "unstable")]
+    pub fn debug_data_shutdown(&mut self) {
```

Comment:
I realized right as I was finishing up the checklist in that issue that coming up with a unit test would be harder, since there would be no real way to externally test it _works_. So I appreciate trying to come up with one 😅  And that this existing to begin with as "unstable" for testing purposes seems fine.

I meant to file a follow-up issue after, adding debug data to a specific GOAWAY case, which in that once a test exists, we could remove this stuff, I suppose?

Reply:
Adding the test wasn't too hard, to be honest. It was just copying what was already done and modifying as needed. The hardest part was figuring out _where_ the test should be, and how to run it, and that didn't take long either.

As soon as an actual use of debug data is added I agree that all of that stuff should be ripped out again, in favor of code paths